### PR TITLE
OPENEUROPA-2872: Create Facts and figures paragraph.

### PR DESCRIPTION
**Pull request by 22Alexandra**
### Description

Create Facts and figures paragraph.

### Change log

- Added:
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh

```

### CODE REVIEW

File: oe_paragraphs.post_update.php [hunk 1]
```diff
@@ -129,3 +129,55 @@ function oe_paragraphs_post_update_10004(array &$sandbox): void {
...
+  $paragraphs = [
+    'paragraphs.paragraphs_type.oe_fact',
+    'paragraphs.paragraphs_type.oe_facts_figures',
+  ];
+  foreach ($paragraphs as $paragraph) {
+    $paragraph_storage->create($storage->read($paragraph))
+      ->save();
+  }
+
+  // Add the fields to paragraphs.
```
Comment (imanoleguskiza) [inline oe_paragraphs.post_update.php]: We are not only adding fields, we are also setting form and view displays.

File: tests/features/facts-figures.feature [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    # Check the fields of the referenced Fact paragraph.
+    And the following fields should be present "Figure, Icon, Description, Label"
+
+    When I press "Save"
+    Then I should see the following error messages:
+      | error messages           |
+      | Label field is required  |
+      | Figure field is required |
+
+    When I fill in "Title" with "Facts and figures" in the 1st "Facts and figures" paragraph
```
Comment (imanoleguskiza) [inline tests/features/facts-figures.feature]: Let's add one more fact for completeness sake.

File: config/install/core.entity_form_display.paragraph.oe_facts_figures.default.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    - paragraphs.paragraphs_type.oe_facts_figures
+  module:
+    - link
+    - oe_paragraphs
+id: paragraph.oe_facts_figures.default
+targetEntityType: paragraph
+bundle: oe_facts_figures
+mode: default
+content:
+  field_oe_link:
```
Comment (ademarco) [inline config/install/core.entity_form_display.paragraph.oe_facts_figures.default.yml]: Can we move the link at the bottom of the form?

File: config/install/field.field.paragraph.oe_facts_figures.field_oe_paragraphs.yml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  module:
+    - entity_reference_revisions
+id: paragraph.oe_facts_figures.field_oe_paragraphs
+field_name: field_oe_paragraphs
+entity_type: paragraph
+bundle: oe_facts_figures
+label: Fact
+description: 'Add between 1 to 6 Facts.'
+required: true
+translatable: true
```
Comment (ademarco) [inline config/install/field.field.paragraph.oe_facts_figures.field_oe_paragraphs.yml]: Entity reference fields referencing paragraphs should always be set to `translatable: false`. It's something the paragraphs module advocates.

File: config/install/core.entity_form_display.paragraph.oe_facts_figures.default.yml (hunk 3 above)
Comment (ademarco) [reply config/install/core.entity_form_display.paragraph.oe_facts_figures.default.yml]: Please ignore me. :smile:

File: oe_paragraphs.info.yml [hunk 5]
```diff
@@ -139,6 +154,20 @@ config_devel:
...
+    - field.storage.paragraph.field_oe_limit
+
+    # Plain long text (long text)
+    - field.storage.paragraph.field_oe_plain_text_long
+
+    # Social media links (typed link)
+    - field.storage.paragraph.field_oe_social_media_links
+
+
+
```
Comment (ademarco) [inline oe_paragraphs.info.yml]: Leave only one empty line.

File: oe_paragraphs.post_update.php [hunk 6]
```diff
@@ -129,3 +129,55 @@ function oe_paragraphs_post_update_10004(array &$sandbox): void {
     $display->save();
   }
 }
+
+/**
+ * Installs Facts and figures and Fact paragraphs.
+ */
+function oe_paragraphs_post_update_10006(array &$sandbox) {
```
Comment (ademarco) [inline oe_paragraphs.post_update.php]: Should be `oe_paragraphs_post_update_10005()`

File: config/post_updates/10005/core.entity_form_display.paragraph.oe_fact.default.yml [hunk 7]
```diff
@@ -0,0 +1,47 @@
+langcode: en
```
Comment (ademarco) [inline config/post_updates/10005/core.entity_form_display.paragraph.oe_fact.default.yml]: This file seems to be different from what stored in config/install, see diff below:

```diff
<     weight: 0
---
>     weight: 1
22c22
<     weight: 3
---
>     weight: 2
30c30
<     weight: 2
---
>     weight: 3
38c38
<     weight: 1
---
>     weight: 0
```

Comment (ademarco) [reply config/post_updates/10005/core.entity_form_display.paragraph.oe_fact.default.yml]: IN fact the fields order seem messed up when upgrading from master to this branch:

image

Comment (ademarco) [reply config/post_updates/10005/core.entity_form_display.paragraph.oe_fact.default.yml]: This is what I get from a clean installation:

image

File: oe_paragraphs.post_update.php [hunk 8]
```diff
@@ -129,3 +129,70 @@ function oe_paragraphs_post_update_10004(array &$sandbox): void {
...
+function oe_paragraphs_post_update_10006(array &$sandbox): void {
+  $storage = new FileStorage(drupal_get_path('module', 'oe_paragraphs') . '/config/post_updates/10006');
+  $config_manager = \Drupal::service('config.manager');
+  $entity_manager = \Drupal::entityTypeManager();
+  $config_record = $storage->read('field.storage.paragraph.field_oe_icon');
+  $entity_type = $config_manager->getEntityTypeIdByName('field.storage.paragraph.field_oe_icon');
+  $entity_storage = $entity_manager->getStorage($entity_type);
+  $entity = $entity_storage->load($config_record['id']);
+  $entity_storage->updateFromStorageRecord($entity, $config_record);
+  $entity->save();
```
Comment (ademarco) [inline oe_paragraphs.post_update.php]: Let's just change those 3 values in the field configuration so we don't have to maintain a full copy of it in the 10006 directory.

**Pull request merged**